### DISCUSSION TITLE
redis与qcluster依赖冲突

### DISCUSSION THREAD

**Discussion by nick2wang** [hhyo/Archery#1966] [General]
> redis集群模式下执测试连接报：module 'redis' has no attribute 'cluster' archery版本1.8.5

>查看了一下Python Redis库版本，3.5.4，看了[文档，需要4.1.0开始才支持](https://pypi.org/project/redis/4.1.0/)。
自己下载镜像，升了个级，目前Redis cluster能用了，配置一个节点。

_Originally posted by @zhaoge26 in https://github.com/hhyo/Archery/issues/1392#issuecomment-1315582620_

redis集群模式依赖redis==4.1.0，与qcluster==1.3.9的依赖冲突（实际使用上不影响，但是批量安装时会中断，导致构建镜像ci报错；qcluster项目已经一年多没更新）
```
ERROR: Cannot install -r requirements.txt (line 12) and redis==4.1.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested redis==4.1.0
    django-q 1.3.9 depends on redis<4.0.0 and >=3.5.3

```

目前只能后期手动升级redis module的版本

**Comment by feichenxue** [comment]
这个如果build镜像就有点麻烦了

**Comment by LeoQuote** [comment]
django-q 提个 pr 看看, 看了下, 项目不活跃了, 有人提过 pr  https://github.com/Koed00/django-q/pull/660 但是没有merge.

可以考虑换了算了 https://github.com/rq/django-rq 可能可以考虑这个 , scheduler 可以考虑换成 https://github.com/rq/rq-scheduler

**Reply by LeoQuote** [reply]
另外也可以改安装脚本, 在 requirements 安装完成之后, 手动安装 redis 新版, `pip install "redis>=4.1.0"`

**Comment by dearsiang** [comment]
Hi 大大

這邊使用docker 安裝archery , 也是遇到 module 'redis' has no attribute 'cluster'，這個錯誤訊息，看了以上解決辦法，登入到archery docker 容器內用pip3 install redis==4.1.0，並重新啟動archery 容器，還是一樣出現module 'redis' has no attribute 'cluster'，請問這邊有少做什麼嗎？或是解決的方法需要修正？

**Reply by dearsiang** [reply]
這邊是使用archery v1.9.1

**Reply by LeoQuote** [reply]
建议升级，最新版解决了这个问题，可以参见最新版代码里的dockerfile
